system/Database/Forge.php

```diff
@@ -302,14 +302,12 @@ public function dropDatabase(string $dbName): bool
      *
      * @return Forge
      */
-    public function addKey($key, bool $primary = false, bool $unique = false)
+    public function addKey($key, bool $primary = false, bool $unique = false, string $keyName = '')
     {
         if ($primary) {
-            foreach ((array) $key as $one) {
-                $this->primaryKeys[] = $one;
-            }
+            $this->primaryKeys = ['fields' => (array) $key, 'keyName' => $this->db->protectIdentifiers($keyName)];
```

Comment:
Why do we need to escape `keyName` here?
I think it is better to escape once when generating SQL statements.
That is right before outputting as possible.

Reply:
Your right, It actually was going to interfere with checking if the key exists in fields. I changed it..